api/v1alpha1/conditions.go

```diff
@@ -0,0 +1,89 @@
+package v1alpha1
+
+import v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type conditionsBuilder struct {
+	t       string
+	Status  v1.ConditionStatus
+	reason  string
```

Comment:
Any particular reason why reason is non exported whereas Status and Message are. Also, can we rename t to type/Type itself?